### ISSUE TITLE
After converting, coreML model is removing some factors

### ISSUE THREAD

**Issue by Shradha131** [onnx/onnx-coreml#317]
I converted my onnx model to coreML using onnx-coreml but my converted coreML model is not having dropout factor, which was initially present in my onnx model.
Further, this tool also removed momentum and affine from my batch normalisation.

Is there any way, I can convert my onnx model to coreML retaining these factors.

**Comment by aseemw** [comment]
This is on purpose. CoreML does not support dropout or momentum etc in BN as they are training related layers/parameters and CoreML is an inference only engine. 

Why do need these in your CoreML graph? Is your model using these features during inference?

**Comment by Shradha131** [comment]
Yes, my model is using these feature as dropout factor, momentum and affine during inference.

**Comment by aseemw** [comment]
Okay. 
May I ask what kind of model it is? (if its not open source, ignore this question :)).
Having dropout means, it gives a non deterministic output each time you invoke it? 

In that case, you would need to use the custom layers in CoreML. There are some pointers in the readme of this repo as to which extra arguments to pass to the converter and examples in the unit tests. 

Here is a nice [blogpost](http://machinethink.net/blog/coreml-custom-layers/) about it.
